integration-tests/ccip-tests/actions/ccip_helpers.go

```diff
@@ -1648,44 +1648,48 @@ func (sourceCCIP *SourceCCIPModule) IsRequestTriggeredWithinTimeframe(timeframe
 
 // IsPastRequestTriggeredWithinTimeframe determines the average block time and calculates the block numbers
 // within the specified timeframe. It then uses FilterCCIPSendRequested to identify the past events.
-func (sourceCCIP *SourceCCIPModule) IsPastRequestTriggeredWithinTimeframe(ctx context.Context, timeframe *commonconfig.Duration) (*time.Time, error) {
+func (sourceCCIP *SourceCCIPModule) IsPastRequestTriggeredWithinTimeframe(
+	ctx context.Context,
+	timeframe *commonconfig.Duration,
+) (latestEvent *types.Log, err error) {
 	if timeframe == nil {
 		return nil, nil
 	}
 	//var foundAt *time.Time
-	latestBlock, err := sourceCCIP.Common.ChainClient.LatestBlockNumber(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("error while getting latest source block number. Error: %w", err)
+	latestBlock, err1 := sourceCCIP.Common.ChainClient.LatestBlockNumber(ctx)
+	if err1 != nil {
+		return nil, fmt.Errorf("error while getting latest source block number. Error: %w", err1)
 	}
-	avgBlockTime, err := sourceCCIP.Common.ChainClient.AvgBlockTime(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("error while getting average source block time. Error: %w", err)
+	avgBlockTime, err1 := sourceCCIP.Common.ChainClient.AvgBlockTime(ctx)
+	if err1 != nil {
+		return nil, fmt.Errorf("error while getting average source block time. Error: %w", err1)
 	}
 	filterFromBlock := latestBlock - uint64(timeframe.Duration()/avgBlockTime)
 
-	onRampContract, err := evm_2_evm_onramp.NewEVM2EVMOnRamp(common.HexToAddress(sourceCCIP.OnRamp.EthAddress.Hex()),
+	onRampContract, err1 := evm_2_evm_onramp.NewEVM2EVMOnRamp(common.HexToAddress(sourceCCIP.OnRamp.EthAddress.Hex()),
 		sourceCCIP.Common.ChainClient.Backend())
-	if err != nil {
-		return nil, fmt.Errorf("error while on ramp contract. Error: %w", err)
+	if err1 != nil {
+		return nil, fmt.Errorf("error while on ramp contract. Error: %w", err1)
 	}
-	iterator, err := onRampContract.FilterCCIPSendRequested(&bind.FilterOpts{
+	iterator, err1 := onRampContract.FilterCCIPSendRequested(&bind.FilterOpts{
 		Start: filterFromBlock,
 	})
-	if err != nil {
-		return nil, fmt.Errorf("error while filtering CCIP send requested starting block number: %d. Error: %w", filterFromBlock, err)
+	if err1 != nil {
+		return nil, fmt.Errorf("error while filtering CCIP send requested starting block number: %d. Error: %w", filterFromBlock, err1)
 	}
 	defer func() {
-		_ = iterator.Close()
+		err = iterator.Close()
```

Comment:
Can just change this to `interatorErr` and log it if it's not nil instead of changing the outer err.

```suggestion
		iterErr = iterator.Close()
		if iterErr != nil {
		     sourceCCIP.Logger.Error().Err(err).Msg("Error closing iterator")
		 }
```

Reply:
Updated